td/views.py

```diff
@@ -56,85 +63,141 @@ def get_todo_list_items(request):
-    If no then it creates new entry for him in 'Users' Postgres table and after
-    that appends to the 'Items' collection his todo list item from POST json
-    body in Mongo database.
+    function checks if user's id exists in the Postgres database table 'Users'.
+    If no then raises HTTPUnauthorized.
 
     :param request: instance-object which represents HTTP request.
-    :type: pyramid.request.Request
+    :type request: pyramid.request.Request
     :returns: Response instance with 'OK' str body to indicate a success.
     :rtype: pyramid.response.Response
 
     """
     settings = request.registry.settings
     ip = request.client_addr
-    logger.debug("User with ip %s is trying to add new item with POST "
-                 "request on /api/add_todo_list_items",
-                 ip)
-
-    db = settings.db
-    user_id = db.select_one("id", "Users", "ip='%s'" % ip)
-    if user_id is None:
-        logger.debug("User with such ip does not exist in the database, "
-                     "creating new entry for him in Users table")
-        db.insert("Users", "ip", ip)
-        user_id = db.select_one("id", "Users", "ip='%s'" % ip)[0]
-    # Can't use .get here because of errror: TypeError:
-    # 'builtin_function_or_method' object has no attribute '__getitem__'
-    mongo_creds = settings.mongo_creds
+    login_name = authenticated_userid(request)
+    logger.debug("User with ip %s and username '%s' is trying to add new item"
+                 "with POST request on /api/add_todo_list_items",
+                 ip, login_name)
+
+    db = settings["db"]
+    user_int_id = db.select_one("id", "Users", "username='%s'" % login_name)
+    if user_int_id is None:
+        return HTTPUnauthorized()
+    # extract id integer from tuple like (2,)[0] -> 2
+    user_int_id = user_int_id[0]
+    mongo_creds = settings["mongo_creds"]
     client = pymongo.MongoClient(mongo_creds['host'],
                                  int(mongo_creds['port']))
-    db = client[mongo_creds['db_name']]
-    items_collection = db.Items
+    mongo_db = client[mongo_creds['db_name']]
+    items_collection = mongo_db.Items
     items_collection.insert_one({"item": request.json_body["item"],
-                                 "owner_id": user_id})
+                                 "owner_id": user_int_id})
 
     logger.debug("Successfully added item '%s' to the database.",
                  request.json_body["item"])
 
     return Response("OK")
+
+
+def get_login_page(request):
+    """Return static/login.html at request on /login url.
+
+    :param request: instance-object which represents HTTP request.
+    :type request: pyramid.request.Request
+    :returns: object that is used to serve a file from static/login.html.
+    :rtype: pyramid.response.FileResponse
+    """
+    logger.debug("Get request for resource at /login and replied with "
+                 "file static/login.html")
+    abs_path_to_base = "".join([os.path.dirname(__file__),
```

Comment:
You don't need this. You can use: 

`os.path.join(os.path.dirname(__file__), "static", "login.html")`